### ISSUE TITLE
Seasons Not Melting Snow/Ice By Default Settings

### ISSUE THREAD

**Issue by jredfox** [TheWeatherPony/TheSeasonsMod#44]
Priority: High
Version: beta9
Images Snow: http://imgur.com/a/0OMMn
Images Ice: http://imgur.com/a/KrFYi
Also made sure that these biomes were in summer as well. 

Description:
Snow/Ice doesn't melt by default settings when last time it did melt. I then tried to see if it was the seasons.dat file which PitchBright gave me his and it seemed to have fixed the issue. This proves it is the default settings not allowing it to melt snow/ice when seasons allows it to.  

_Unseccesfull_
1. created world
2. /seasons set sesoasons winter 
3. /toggledownfall
4. /seasons set seasons spring
5. /toggledownfall
6. made sure it was day
7. snow didn't melt so I decided to do /seasons set seasons summer
8. after /seasons set season summer didn't work. no snow/ice melted

**succesfull**
1. created world
2. closed mc
3. deleted seasons.dat and replaced it with PitchBrights configured seasons.dat
4. launched mc and logged in the world
5. seasons was already set to spring so it started melting. I don't know how he configured it but, it melted slower in summer.
6. concluded that this was the default seasons.dat not configed by default for snow to melt.

Conclusion:
seasons.dat was not configured right so the snow/ice didn't melt by default regardless weather or not it was summer or spring. If the settings are changed it will melt.  I also noticed in the deafault nbt tag there was **spring defrost 0 by default** so I think that could have been the issue However I am not sure.

**Comment by TheWeatherPony** [comment]
There is a legacy setting in the NBT configs that was previously enabled for debugging purposes, and which PitchBright currently uses because his server is centered around an Extreme Hills biome, which is never very warm - this setting is named "SpringDefrost" and was added and used by default long before any temperature-based melting was added. This setting acts to melt snow and ice extremely quickly, in addition to the slower temperature-based melting.

The temperature based melting for snow works as such:
- When snow gets ticked, there is a (NBT->snowLogic->meltAttempt) out of 1 chance that the melting logic tries to run.
- If the temperature is less than .15, the melt attempt fails, and it does not melt. (since snow/ice form at that temperature)
- There is then a chance the melt will occur. I keep tweaking this chance, but it is currently ((temperature - .15) ^3) out of ((.65)^3 x 1.65) - this makes it guaranteed to melt at a temperature of .918 and above, assuming the current attempt passes the meltAttempt chance.
- If it is determined that the melt should occur, one snow tile layer is removed from the top of the stack which the current snow tile block belongs to.
- This process repeats at random intervals on random blocks by the vanilla block update code whenever there is snow to do it on

**Looking through the code, there is currently a bug where ice does not experience temperature-based melting. This will be fixed for the next version.**

**Comment by jredfox** [comment]
I waited 3 days in the summer and it didn't melt.

**Comment by PitchBright** [comment]
snow, or ice?

**Comment by jredfox** [comment]
both. I shared screens with you after a while of it not melting .
The_WeatherPony
 I noticed ice didn't melt in the desert I was going to report this but, I can see you have already found that out. snow didn't appear to be melting and it seemed like it was the seasons.dat because, I used PitchBrights And it was melting more like it use to . I know you turned it down a bit but, how long is it suppose to take to melt. I can test this out agian to see if anything else comes up today.

**Comment by jredfox** [comment]
nope still didn't work and this time I tried a hot biome.

**Comment by jredfox** [comment]
Images link was added and I can rule out tempiture since A seasons.dat fixes it B tested it in a flat nether world and in the nether dimension and the snow didn't melt. http://imgur.com/a/0OMMn . The image url is also at the top

**Comment by TheWeatherPony** [comment]
When I go and test it for myself, snow melts properly. I've already made changes in that section of the code for the next version, including the formulas involved, so such a problem may have been fixed indirectly, already. Regardless, it is working for the next version.
Also, I've made it so that ice blocks will melt properly in the next version.

The temperature melting formula for the next version is linear, rather than cubic, making it much easier to follow. Ice runs the logic twice, to make it less susceptible to melting than the snow tiles.
The craftable snow blocks will remain unaffected by seasonal heat, as an intended feature for builders. This may be readdressed for MC1.9, which adds igloos.
